lua/rustaceanvim/rust_analyzer.lua

```diff
@@ -5,18 +5,33 @@ local os = require('rustaceanvim.os')
 ---@class rustaceanvim.rust-analyzer.ClientAdapter
 local M = {}
 
+---@class GetClientFilter
+---@field filter? vim.lsp.get_clients.Filter
+---@field rustc_target? string Optional cargo target to filter rust-analyzer clients
+
 ---@param bufnr number | nil 0 for the current buffer, `nil` for no buffer filter
----@param filter? vim.lsp.get_clients.Filter
+---@param filter? GetClientFilter
 ---@return vim.lsp.Client[]
 M.get_active_rustaceanvim_clients = function(bufnr, filter)
   ---@type vim.lsp.get_clients.Filter
-  filter = vim.tbl_deep_extend('force', filter or {}, {
+  local client_filter = vim.tbl_deep_extend('force', filter and filter.filter or {}, {
     name = 'rust-analyzer',
   })
   if bufnr then
-    filter.bufnr = bufnr
+    client_filter.bufnr = bufnr
   end
-  return vim.lsp.get_clients(filter)
+  local clients = vim.lsp.get_clients(client_filter)
+  clients = vim.tbl_filter(function(client)
+    local rust_settings = client.config and client.config.settings and client.config.settings['rust-analyzer']
+    local cargo_target = rust_settings and rust_settings.cargo and rust_settings.cargo.target
+    -- If no filter is provided or filter.rustc_target is nil, include the client only if cargo_target exists
+    if not filter or not filter.rustc_target then
+      return cargo_target ~= nil
+    end
+    -- Include the client if cargo_target differs from filter.rustc_target
```

Comment:
```suggestion
```

No need for this comment if we rename the filter param to `exclude_rustc_target`.
I prefer longer, more expressive variable names over comments.